File: yggdrasil/examples/composite_function2/src/model_function.c
#include <stdio.h>

int model_function(bool a, double b, generic_t c,
		   bool* d, double* e,
		   double** f, size_t* f_length) {
  d[0] = (!a);
  e[0] = generic_map_get_double(c, "c1");
  f_length[0] = 3;
  f[0] = (double*)realloc(f[0], 3 * sizeof(double));
  for (int i = 0; i < 3; i++) {
    if (a)
      f[0][i] = b * pow(i, generic_map_get_double(c, "c1"));
    else
      f[0][i] = b * pow(i, generic_map_get_double(c, "c2"));
  }
  return 1;
}

File: yggdrasil/examples/composite_function2/src/model_function.jl
function model_function(a, b, c)
  d = !a
  e = c["c1"]
  f = Array{Float64}(undef, 3)
  for i = 1:3
    if (a)
      f[i] = b * ((i - 1) ^ c["c1"])
    else
      f[i] = b * ((i - 1) ^ c["c2"])
    end
  end
  return d, e, f
end